control/autoware_joy_controller/src/autoware_joy_controller/autoware_joy_controller_node.cpp

```diff
@@ -130,37 +130,41 @@ bool AutowareJoyControllerNode::isDataReady()
   // Joy
   {
     if (!joy_) {
-      ROS_WARN_THROTTLE(1.0, "waiting for joy msg...");
+      RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), std::chrono::milliseconds(1000).count(),
+        "waiting for joy msg...");
       return false;
     }
 
     constexpr auto timeout = 2.0;
-    const auto time_diff = ros::Time::now() - last_joy_received_time_;
-    if (time_diff.toSec() > timeout) {
-      ROS_WARN_THROTTLE(1.0, "joy msg is timeout");
+    const auto time_diff = rclcpp::Clock().now() - last_joy_received_time_;
```

Comment:
We should be using the node time instead of the Clock time since this uses `RCL_SYSTEM_TIME` as the default clock that doesn't play well with simulation time. If you use `this->now()` you can access the node's clock and get the correct time while in simulation as this clock will be associated with the `use_sim_time` parameter. This is in other areas of the code as well

Reply:
Thanks for pointing it out, fixed in all occurrences now

Reply:
No worries, thanks for making the fix.